plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Prism/Withdraw.hs

```diff
@@ -0,0 +1,108 @@
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts   #-}
+{-# LANGUAGE MonoLocalBinds     #-}
+{-# LANGUAGE NamedFieldPuns     #-}
+{-# LANGUAGE NoImplicitPrelude  #-}
+{-# LANGUAGE TemplateHaskell    #-}
+{-# LANGUAGE TypeApplications   #-}
+{-# LANGUAGE TypeFamilies       #-}
+{-# LANGUAGE TypeOperators      #-}
+-- | The withdrawal application that withdraws some funds
```

Comment:
from where?

Reply:
Might be nice to do the exchange withdrawal example too? Which is even simpler: just a way to lock an output in a way that you can only unlock it when presenting a particular token.

Reply:
Yes, I just remembered that we already have a `TokenAccount` contract that does exactly this.